org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolSettings.java

```diff
@@ -85,4 +85,13 @@ public void setMaxItemsComputed(int maxItemsComputed) {
 		this.maxItemsComputed = maxItemsComputed;
 	}
 
+	public void merge(XMLSymbolSettings newSettings) {
+		this.setEnabled(newSettings.isEnabled());
+		String[] newPatterns = newSettings.getExcluded();
+		if (newPatterns != null) {
```

Comment:
in general, you need to perform an array copy, else mutating this setting's array will affect the other one's and vice versa

Reply:
fixed